src/signals/incident/components/form/MapSelectors/Asset/Selector/WfsLayer/StatusLayer/StatusLayer.test.tsx

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2021 Gemeente Amsterdam
+import type { FeatureCollection } from 'geojson'
+
+import { render, screen } from '@testing-library/react'
+import { Map } from '@amsterdam/react-maps'
+
+import type {
+  Feature,
+  FeatureType,
+} from 'signals/incident/components/form/MapSelectors/types'
+import type { AssetSelectValue } from 'signals/incident/components/form/MapSelectors/Asset/types'
+
+import caterpillarsJson from 'utils/__tests__/fixtures/caterpillars.json'
+import { meta, selection } from 'utils/__tests__/fixtures/caterpillarsSelection'
+
+import MAP_OPTIONS from 'shared/services/configuration/map-options'
+import { WfsDataProvider } from 'signals/incident/components/form/MapSelectors/Asset/Selector/WfsLayer/context'
+import withAssetSelectContext, {
+  contextValue,
+} from 'signals/incident/components/form/MapSelectors/Asset/__tests__/withAssetSelectContext'
+import StatusLayer from './StatusLayer'
+
+const assetSelectProviderValue: AssetSelectValue = {
+  ...contextValue,
+  selection: selection[0],
+  meta,
+}
+
+const reportedFeatureType = meta.featureTypes.find(
+  ({ typeValue }) => typeValue === 'reported'
+)
+const checkedFeatureType = meta.featureTypes.find(
+  ({ typeValue }) => typeValue === 'checked'
+)
+
+const statusFeatures = caterpillarsJson.features.filter(
+  (feature) =>
+    Boolean(
+      reportedFeatureType?.isReportedField &&
+        feature.properties['AMS_Meldingstatus'] ===
+          reportedFeatureType?.isReportedValue
+    ) ||
+    Boolean(
+      checkedFeatureType?.isCheckedField &&
+        checkedFeatureType.isCheckedValues?.includes(
+          feature.properties['Registratie']
+        )
+    )
+)
+
+describe('StatusLayer', () => {
+  const withMapCaterpillar = () =>
+    withAssetSelectContext(
+      <Map data-testid="map-test" options={MAP_OPTIONS}>
+        <WfsDataProvider value={caterpillarsJson as FeatureCollection}>
+          {statusFeatures?.length > 0 &&
+            reportedFeatureType &&
+            checkedFeatureType && (
```

Comment:
Why are these conditions required to render the component? This is a test suite and the fixtures should contain the data required to render the unit under test. I would suggest removing these conditions and having a test case that verifies the conditions. The tests currently rely on external conditions that might change. When they change, the tests fail when they shouldn't.

Reply:
I was struggling with the typing earlier. Now I have removed these conditions.